accounts/admin.py

```diff
@@ -155,91 +108,145 @@ def clean_username(self):
-        tvars = {'anonymised': []}
-        tvars['anonymised'].append({'name': info['anonymised']})
+        user_info = obj.profile.get_info_before_delete_user(include_sounds=True, include_other_related_objects=True)
+        user_info['deleted_objects_details'] = {}
+        model_count = {model._meta.verbose_name_plural: len(objs) for
+                       model, objs in user_info['deleted'].model_objs.items()}
+        user_info['deleted_objects_details']['model_count'] = dict(model_count).items()
+        user_info['deleted_objects_details']['logic_deleted'] = user_info['logic_deleted']
+
+        tvars = {'users_to_delete': [], 'type': 'delete_spammer'}
+        tvars['users_to_delete'].append(user_info)
+
+        return render(request, 'accounts/delete_confirmation.html', tvars)
+
+    delete_spammer.label = "Delete spammer"
+    delete_spammer.short_description = "Delete the user and the sounds, mark deleted user as spammer"
+
+    def full_delete(self, request, obj):
+        username = obj.username
+        if request.method == "POST":
+            logger.info('Requested async deletion of user {0} - {1}'.format(obj.id,
+                                                                            FULL_DELETE_USER_ACTION_NAME))
+            gm_client = gearman.GearmanClient(settings.GEARMAN_JOB_SERVERS)
+            gm_client.submit_job("delete_user",
+                                 json.dumps({'user_id': obj.id, 'action': FULL_DELETE_USER_ACTION_NAME}),
+                                 wait_until_complete=False, background=True)
+            messages.add_message(request, messages.INFO,
+                                 'User \'%s\' will be deleted asynchronously including sounds and all of its related'
+                                 'content.' % username)
+            return HttpResponseRedirect(reverse('admin:auth_user_changelist'))
+
+        user_info = obj.profile.get_info_before_delete_user(include_sounds=True, include_other_related_objects=True)
+        user_info['deleted_objects_details'] = {}
+        model_count = {model._meta.verbose_name_plural: len(objs) for
+                       model, objs in user_info['deleted'].model_objs.items()}
+        user_info['deleted_objects_details']['model_count'] = dict(model_count).items()
+        user_info['deleted_objects_details']['logic_deleted'] = user_info['logic_deleted']
+
+        tvars = {'users_to_delete': [], 'type': 'full_delete'}
+        tvars['users_to_delete'].append(user_info)
+
         return render(request, 'accounts/delete_confirmation.html', tvars)
-    delete_preserve_sounds.label = "Soft delete user (preserve sounds)"
-    delete_preserve_sounds.short_description = disable_active_user_preserve_sounds.short_description
 
-    change_actions = ('full_delete', 'delete_include_sounds', 'delete_preserve_sounds', )
+    full_delete.label = "Full delete"
+    full_delete.short_description = 'Completely delete user from db'
```

Comment:
I think this one should have a bit more information. I see the comment in the gearman worker says "WARNING: This functions creates no DeletedSound nor DeletedUser objects". It would be nice to have something similar here, and maybe an indicator about when this should actually be used? Can you give an example of when we might want to use this instead of "Delete User and Sounds"  ?

Reply:
I guess the answer is "ideally we should never use that". So maybe I just remove the option for now and in the advent of an unexpected case that requires it, we can do it using the console or we can re-enable the option.

Reply:
Addressed in f07a66a16c060f948b49ed849b47c04692683f9d (removed the non-necessary option)